src/generators/operationGenerator.ts

```diff
@@ -339,24 +435,39 @@ function getOperationParameterSignatures(
         usedModels.forEach(model => importedModels.add(model));
       }
 
-      return {
+      const newParameter = {
         name: param.name,
         description: param.description,
         type: typeName,
         hasQuestionToken: !param.required
       };
-    });
 
-    // add optional parameter
-    const optionalParameter = getOptionsParameter(
+      // Make sure required parameters are added before optional
+      return param.required ? [newParameter, ...acc] : [...acc, newParameter];
```

Comment:
In order to preserve ordering, we could find the last required parameter already in the list and add the next required element after it. Something like:
```suggestion
      // Make sure required parameters are added before optional
      function findLastRequiredParamIndex(
        params: ParameterWithDescription[]
      ): number {
        for (let i = params.length; i--; ) {
          if (!params[i].hasQuestionToken) {
            return i;
          }
        }
        return -1;
      }
      const newParameterPosition = param.required
        ? findLastRequiredParamIndex(acc) + 1
        : acc.length;
      acc.splice(newParameterPosition, 0, newParameter);
      return acc;
```

Reply:
Thanks!!! I've integrated your suggestion :smile: